data_steward/cdr_cleaner/manual_cleaning_rules/remediate_basics.py

```diff
@@ -279,62 +279,66 @@ def get_query_specs(self):
-            for (domain, table) in [
-                (SURVEY_CONDUCT, SURVEY_CONDUCT),
-                (SURVEY_CONDUCT, SC_EXT),
-                (PERSON, PERSON),
-            ]
+            self._get_query(GENERIC_SANDBOX, SURVEY_CONDUCT, SURVEY_CONDUCT),
+            self._get_query(GENERIC_SANDBOX, PERSON, PERSON)
         ]
 
         delete_queries = [
             self._get_query(GENERIC_DELETE, domain, table)
             for (domain, table) in [
                 (OBSERVATION, OBSERVATION),
-                (OBSERVATION, OBS_EXT),
                 (SURVEY_CONDUCT, SURVEY_CONDUCT),
-                (SURVEY_CONDUCT, SC_EXT),
                 (PERSON, PERSON),
             ]
         ]
 
         insert_queries = [
             self._get_query(INSERT_OBS, OBSERVATION, OBSERVATION),
-            self._get_query(INSERT_OBS_EXT, OBSERVATION, OBS_EXT)
-        ] + [
-            self._get_query(GENERIC_INSERT, domain, table)
-            for (domain, table) in [(SURVEY_CONDUCT, SURVEY_CONDUCT),
-                                    (SURVEY_CONDUCT, SC_EXT), (PERSON, PERSON)]
+            self._get_query(GENERIC_INSERT, SURVEY_CONDUCT, SURVEY_CONDUCT),
+            self._get_query(GENERIC_INSERT, PERSON, PERSON)
         ]
 
-        # mapping tables exist only in combined dataset and its upstream datasets
-        # mapping table for person does not exist in combined dataset
-        if is_combined_dataset(self.dataset_id):
+        # obs_ext and sc_ext exist in combined_release and deid datasets
+        if is_combined_release_dataset(self.dataset_id) or is_deid_dataset(
+                self.dataset_id):
             sandbox_queries.extend([
-                self._get_query(SANDBOX_OBS_MAPPING_EXT, OBSERVATION,
-                                OBS_MAPPING),
-                self._get_query(GENERIC_SANDBOX, SURVEY_CONDUCT, SC_MAPPING)
+                self._get_query(SANDBOX_OBS_MAPPING_EXT, OBSERVATION, OBS_EXT),
+                self._get_query(GENERIC_SANDBOX, SURVEY_CONDUCT, SC_EXT)
             ])
             delete_queries.extend([
-                self._get_query(GENERIC_DELETE, OBSERVATION, OBS_MAPPING),
-                self._get_query(GENERIC_DELETE, SURVEY_CONDUCT, SC_MAPPING)
+                self._get_query(GENERIC_DELETE, OBSERVATION, OBS_EXT),
+                self._get_query(GENERIC_DELETE, SURVEY_CONDUCT, SC_EXT)
             ])
             insert_queries.extend([
-                self._get_query(INSERT_OBS_MAPPING, OBSERVATION, OBS_MAPPING),
-                self._get_query(GENERIC_INSERT, SURVEY_CONDUCT, SC_MAPPING),
+                self._get_query(INSERT_OBS_EXT, OBSERVATION, OBS_EXT),
+                self._get_query(GENERIC_INSERT, SURVEY_CONDUCT, SC_EXT)
             ])
 
-        # person_ext table exists only after DEID
-        elif is_deid_dataset(self.dataset_id):
+        # person_ext table exists only in deid datasets
+        if is_deid_dataset(self.dataset_id):
```

Comment:
Hi, @hiro-mishima - overall this is a good logic addition.  For `person_ext` specifically, it will not exist until `deid_base` or `deid_clean`, and won't exist in `deid`.  Perhaps we should have another function added to `retract_utils.py` like `is_deid_release` that would indicate we are in `deid_base` or `deid_clean`.  Then this new function should be applied in `remediate_basics.py`.

Reply:
This commit should solve the issue: https://github.com/all-of-us/curation/pull/1489/commits/3318e721980ff82c5949fb517012edb6dfa5c98d#diff-638190b97f9a117d7f9c2a0fc16996cb1608d05f353b383919366a8b7aec007fR319

Reply:
I also added a new test case here to confirm the script behaves differently with `DEID` and `DEID_BASE/CLEAN`: https://github.com/all-of-us/curation/pull/1489/commits/3318e721980ff82c5949fb517012edb6dfa5c98d#diff-3ab4ffcf7d0e4d36b4bbabcc4e851831d5b87bead7e57a98844a2d6aca999fd8R612